tests/js/spec/views/settings/projectSampling/modal.spec.tsx

```diff
@@ -0,0 +1,580 @@
+
+    expect(saveMock).toHaveBeenLastCalledWith(
+      '/projects/org-slug/project-slug/',
+      expect.objectContaining({
+        data: {
+          dynamicSampling: {
+            rules: [rule],
+          },
+        },
+      })
+    );
+
+    // Transaction rules panel is updated
+    expect(screen.getByText('Legacy Browser')).toBeInTheDocument();
+    for (const legacyBrowser of legacyBrowsers) {
+      const {title} = LEGACY_BROWSER_LIST[legacyBrowser];
+      expect(screen.getByText(title)).toBeInTheDocument();
+    }
+  });
+
+  it('custom tag condition', async function () {
+    const rule = {
+      condition: {
+        inner: [{name: 'event.tags.user', op: 'glob', value: ['david']}],
+        op: 'and',
+      },
+      id: 0,
+      sampleRate: 0.15,
+      type: 'transaction',
+    };
+
+    MockApiClient.addMockResponse({
+      url: '/organizations/org-slug/tags/user/values/',
+      method: 'GET',
+      body: [{value: 'david'}],
+    });
+    const saveMock = MockApiClient.addMockResponse({
+      url: '/projects/org-slug/project-slug/',
+      method: 'PUT',
+      body: TestStubs.Project({
+        dynamicSampling: {
+          rules: [rule],
+          next_id: 43,
+        },
+      }),
+    });
+
+    renderComponent({ruleType: SamplingRuleType.TRANSACTION});
+
+    // Open Modal
+    await openSamplingRuleModal(screen.getByText('Add Rule'));
+
+    // Uncheck tracing checkbox
+    userEvent.click(screen.getByRole('checkbox'));
+
+    // Click on 'Add condition'
+    userEvent.click(screen.getByText('Add Condition'));
+
+    // Select Custom Tag
+    userEvent.click(screen.getByText('Add Custom Tag'));
+
+    // Type into tag field
+    userEvent.paste(screen.getByLabelText('Search or add a tag'), 'user');
+
+    // Click on the suggested option
+    userEvent.click(screen.getByTestId('user'));
+
+    // Type into tag value field
+    userEvent.paste(screen.getByLabelText('Search or add tag values'), 'david');
+
+    // Click on the suggested option
+    userEvent.click(screen.getByTestId('david'));
+
+    // Fill sample rate field
+    userEvent.paste(screen.getByPlaceholderText('\u0025'), '15');
+
+    // Click on 'Add condition'
+    userEvent.click(screen.getByText('Add Condition'));
+
+    // Custom tag condition is added to the conditions dropdown
+    expect(screen.getByText('user - Custom')).toBeInTheDocument();
+
+    // Click on save button
+    userEvent.click(screen.getByLabelText('Save Rule'));
+
+    // Close Modal
+    userEvent.click(screen.getByLabelText('Close Modal'));
+    await waitForElementToBeRemoved(() =>
+      screen.queryByText('Add Transaction Sampling Rule')
+    );
+
+    expect(saveMock).toHaveBeenLastCalledWith(
+      '/projects/org-slug/project-slug/',
+      expect.objectContaining({
+        data: {
+          dynamicSampling: {
+            rules: [rule],
+          },
+        },
+      })
+    );
+  });
+
+  it('does not let you save without permissions', async function () {
```

Comment:
Adds coverage for feature introduced here: https://github.com/getsentry/sentry/pull/34989